llvm/lib/Analysis/Delinearization.cpp

```diff
@@ -714,7 +895,140 @@ bool llvm::tryDelinearizeFixedSizeImpl(
   if (!SrcGEP)
     return false;
 
-  getIndexExpressionsFromGEP(*SE, SrcGEP, Subscripts, Sizes);
+  // When flag useGEPToDelinearize is false, delinearize only using array_info.
+  if (!useGEPToDelinearize) {
```

Comment:
Could you make "delinerize-from-GEP" and "delinerize-from-global-type" each boolean enable options (like `UseFixedSizeArrayHeuristic`). So when "delinerize-from-global-type", it can still fall back to GEP delinearization? This way, we can prepare GEP removal by just switch off that "delinerize-from-GEP" flag. We also do not need to update all tests at once, but can be done incrementally.